### ISSUE TITLE
double backslash required to escape enhanced text control characters

### ISSUE THREAD

**Issue by d-lamb** [drzowie/PDL-Graphics-Gnuplot#51]
It seems that two backslashes are required to properly escape enhanced text control sequences, even though the documentation says only 1 backslash is required. Is that an implementation bug or a documentation bug?

~~~~perl
use PDL::Graphics::Gnuplot;
$w=gpwin("wxt");
$x=xvals(10);
$y=$x*$x;
$l = 'some_plain_text';
$w->plot({le=>$l},$x,$y); ##subscripts 'p' and 't' as expected.
$l=~s/_/\\_/g; print $l;  ##prints 'some\_plain\_text'
$w->plot({le=>$l},$x,$y); ##subscripts 'p' and 't' even though \escaped
$l=~s/\\/\\\\/g; print $l; ##prints 'some\\_plain\\_text'
$w->plot({le=>$l},$x,$y); ##now the underscores show up as intended.
~~~~

**Comment by drzowie** [comment]
Heh.  This is unintended!  It is probably due to some double-quote string interpolation going on in the processing.  Nice catch.

**Comment by drzowie** [comment]
I had a look at this using tee.  Try your example with "$w->option(tee=>'nobinary')".  You'll see that the single backslashes are being passed correctly to gnuplot itself, which is not handling them correctly.
I'm filing a gnuplot bug now.

**Comment by drzowie** [comment]
OK, I filed a bug with Gnuplot, they tell me that type of parsing is "a feature, not a bug" for their double-quote treatment.  Apparently there are two levels of indirection there:  double-quote parsing, *followed* by enhanced-text parsing.

I consider it broken, but given that they're sticking by their guns we can hopefully rely on it remaining that way.  I inserted a backslash-escape wrapper around string emission, so your example will work, and also other things like (e.g.)  C<title=>"foo\nbar"> do the expected thing.  ("foo\nbar" gets turned into a two-line string by Perl, and then back into a one-line string by the escape wrapper, and then finally forward into a two-line title by gnuplot).
